pkg/audit/disqualification_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+	"storj.io/storj/uplink"
+)
+
+func TestDisqualifiedNodesGetNoDownload(t *testing.T) {
+
+	// - uploads random data
+	// - mark a node as disqualified
+	// - check we don't get it when we require order limit
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		disqualifiedNode := planet.StorageNodes[0]
+		uplinkNode := planet.Uplinks[0]
+
+		err := satellite.Audit.Service.Close()
+		require.NoError(t, err)
+
+		testData := make([]byte, 8*memory.KiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = uplinkNode.UploadWithConfig(ctx, satellite, &uplink.RSConfig{
```

Comment:
I guess you are uploading with custom RS config to be sure that `planet.StorageNodes[0]` gets a piece.

This is not really required. Below you get the pointer of the uploaded file. You can use `pointer.GetRemote().GetRemotePieces()[0].NodeId` to get a storage node that has a piece for sure. Then you can disqualify this storage node.